src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -32,65 +34,65 @@ private static void setupPresetTests(Application application) {
 				DriverFeature.getDriverManager());
 		SelectTestFigure2OptionListener selectTestFigure2OptionListener = new SelectTestFigure2OptionListener(
 				DriverFeature.getDriverManager());
-
+		
 		application.addTest("Figure Joe 1", selectTestFigureOptionListener);
 		application.addTest("Figure Joe 2", selectTestFigure2OptionListener);
-
+		
 	}
-
+	
 	/**
 	 * Setup test using driver commands in context.
 	 *
 	 * @param application Application context.
 	 */
 	private static void setupCommandTests(Application application) {
 		application.addTest("Load secret command", new SelectLoadSecretCommandOptionListener());
-
+		
 		application.addTest("DriverCommandVisitor test DriverCommand", new SelectSingleCommandVisitorTestListener());
 		application.addTest("DriverCommandVisitor test ICompoundCommand", new SelectCompoundCommandVisitorTestListener());
 		
 		application.addTest("Load and copy secret command", new SelectCopySecretCommand());
 	}
-
+	
 	/**
 	 * Setup driver manager, and set default Job2dDriver for application.
 	 *
 	 * @param application Application context.
 	 */
 	private static void setupDrivers(Application application) {
-		Job2dDriver loggerDriver = new LoggerDriver();
-
+		Job2dDriver loggerDriver = new DriverUsageMonitor(new LoggerDriver());
```

Comment:
You should not force the decorator in that way. If anyone else adds a new driver your monitor will not work. You can add your own driver combination without changing existing ones.
LATER you can provide some generic mechanism, taking the advantage from "Observer for a driver changes events" task.